### ISSUE TITLE
Cannot update tag of a recovery service managed group

### ISSUE THREAD

**Issue by karataliu** [Azure/azure-libraries-for-net#364]
Code
```
await group.Update().WithTag("a", "b").ApplyAsync();
```
Error
```
The managed by property of the resource group cannot be changed from its current value 'subscriptions/{id}/providers/Microsoft.RecoveryServices/'.
```

If using REST api manually (excluding 'managedBy' property), it will succeed.

**Comment by hovsepm** [comment]
@karataliu What is the type of `group` object? Could you also provide SDK version?

**Comment by karataliu** [comment]
@hovsepm 
group is of `Microsoft.Azure.Management.ResourceManager.Fluent.IResourceGroup` type. 
SDK: `<PackageReference Include="Microsoft.Azure.Management.ResourceManager.Fluent" Version="1.11.1" />`

The resource group is a special one create by azure back service, name is 'AzureBackupRG_*'

**Comment by hovsepm** [comment]
@karataliu I'm unable to repro it on my end with just a generic RG creation and update. 

```
var resourceGroup = resourceManager.ResourceGroups.Define(rgName)
    .WithRegion(Region.USEast2)
    .Create();

var updated = resourceGroup.Update()
    .WithTag("a", "b")
    .Apply();

Assert.NotNull(updated.Tags);
Assert.Equal(1, updated.Tags.Count);
```

This looks like a back-end service issue. Are you hitting this while working with AzureBackup service?

**Comment by karataliu** [comment]
Yes for sure it works well with a normal resource group.

If you're using azure backup service in some subscription, that service will create a resource group in your subscription with name like 'AzureBackupRG_eastasia_1'. 

The SDK will fail to edit tag on that resource group.

**Comment by karataliu** [comment]
Seems this issue exists with any group with 'managedBy' property.
Found a easy way to repro:
1. Use resources.azure.com to create a resource group, and manually specify 'managedBy' with any value.
2. Use previous mentioned code to update tag for the resource group

**Comment by tylerd** [comment]
The same issue occurs when updating the tags for a resource group managed by Azure Kubernetes Service (AKS)

Without going into detail …

The ResourceManager fluent library uses `api-version=2017-05-10` -- as set in the [ResourceManagementClient class here](https://github.com/Azure/azure-libraries-for-net/blob/master/src/ResourceManagement/ResourceManager/Generated/ResourceManagementClient.cs)
The `managedBy` property is included in the `2018-05-01` api version (and possibly versions prior)

In summary … the code generation config file needs to be updated to generate the Fluent sdk for api-version 2018-05-01 --- specifically `tag: package-resources-2018-05`
[as per this README](https://github.com/Azure/azure-rest-api-specs/tree/master/specification/resources/resource-manager)

**Comment by tylerd** [comment]
If anyone is experiencing this issue, you can either use the [core Management SDK](https://www.nuget.org/packages/Microsoft.Azure.Management.ResourceManager) (which is pretty much un-documented …)
Or I believe you can use the Azure PowerShell Module to update the tags, because it uses the updated Management SDK

**Comment by karataliu** [comment]
Still repros:

```
var s1 = SdkContext.AzureCredentialsFactory.FromServicePrincipal(appid, appsec, tenantid, AzureEnvironment.AzureGlobalCloud);
var rgs = ResourceManager.Authenticate(s1).WithSubscription(subid).ResourceGroups;
var group = rgs.GetByName(rg);
await group.Update().WithTag("a", "b").ApplyAsync();
```
error:
`Unhandled exception. Microsoft.Rest.Azure.CloudException: The managed by property of the resource group cannot be changed from its current value `

`<PackageReference Include="Microsoft.Azure.Management.ResourceManager.Fluent" Version="1.32.0" />`
